index.js

```diff
@@ -90,3 +111,11 @@ module.exports = SambaClient;
 function wrap(str) {
   return '\'' + str + '\'';
 }
+
+function checkForErrors(err, allOutput, defaultVal, cb) {
```

Comment:
this can actually run synchronously without a callback. Then in the function above we should return if there is an error. Otherwise the callback could be called but the function still would continue executing.